Lib/test/test_long.py

```diff
@@ -1351,35 +1350,14 @@ def test_shift_bool(self):
                 self.assertEqual(type(value >> shift), int)
 
     def test_as_integer_ratio(self):
-        tests = [10, 0, -10, 1]
+        class myint(int):
+            pass
+        tests = [10, 0, -10, 1, sys.maxsize + 1, True, False, myint(42)]
```

Comment:
Question: what about the `enum.IntEnum` tests? (I assume those can be derived from an int subclass but I just want to confirm).